HyperPlay-Gaming/hyperplay-desktop-client
# [UX] Improve Compatibility layer setup and handling

This PR does several things like:

- Adds a warning on the install dialog when installing a non-native game so the person is aware that might have issues or the game not even work
- Adds a warning when launching non-native games as well. The 'Dont show again' button stores the option per game, not globally. This is intentional since most people won't check if the game is native or not and the result also depends on the game, hardware, system, etc.
- Adds a way of checking if macOS is updated to Sonoma or later so we can enable or disable GPTK automatically.
- Adds a check to see if Rosetta is installed on mac M1+ and instructions on how to install it if not.
- Use Wine-crossover if macOS ir not updated.
- If GPTK is selected to run a game, show a warning about it and an option to use Wine instead.
- Even if GPTK is available, if not on latest macOS it will download Wine.
- Fix Winetricks download being skipped on macOS and also uses main URL for it.


### How to test

This one is tricky to test if you have updated your Mac to Sonoma I believe.
But the screen below should appear for all Windows games on both macOS and Linux:
The other changes would require me to change the logic to check for a version of Mac that does not exist yet or with a Mac using Ventura or below.

Warning on Installing non-native game:
<img width="781" alt="Screenshot 2024-02-26 at 20 22 01" src="https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/26871415/33db0c53-a3ce-4fee-8ee3-899e2bca4201">

Warning when launching non-native game:
<img width="773" alt="Screenshot 2024-02-26 at 20 38 45" src="https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/26871415/636ee975-50b6-4f01-93b4-730ed2ac79b4">

Warning on Rosetta not available:
<img width="271" alt="Screenshot 2024-02-27 at 10 51 09" src="https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/26871415/358221fb-5ab3-40e6-9f03-83ae360e9041">


---

Use the following Checklist if you have changed something on the Backend or Frontend:
